src/node_constants.cc

```diff
@@ -800,6 +800,15 @@ void DefineCryptoConstants(Local<Object> target) {
     NODE_DEFINE_CONSTANT(target, OPENSSL_VERSION_NUMBER);
 #endif
 
+    // When defined, its defined without a value, so redefine it as 1 or 0.
+#ifdef OPENSSL_NO_SSL_TRACE
+# undef OPENSSL_NO_SSL_TRACE
+# define OPENSSL_NO_SSL_TRACE 1
+#else
+#  define OPENSSL_NO_SSL_TRACE 0
+#endif
+    NODE_DEFINE_CONSTANT(target, OPENSSL_NO_SSL_TRACE);
```

Comment:
Is it even worth exposing `OPENSSL_NO_SSL_TRACE`? We don't expose any of the other `OPENSSL_NO_*` defines either.

I'd remove it if it's just for feature detection in the test. You could make `TLSWrap::EnableTrace()` return a boolean or throw an exception and use that for feature detection.

Reply:
Its annoyingly more complex than it might seem. The return value of enableTrace isn't seen on the server side (its called before the user code gets the TLSSocket), and since the tests have to call them in a sub-process to catch stderr, it makes a simple test skip more fragile than needed. I'll move these to the internal binding so they aren't visible in our API.